crates/sui-graphql-rpc/src/extensions/query_limits_checker.rs

```diff
@@ -147,6 +157,48 @@ impl<'a> LimitsTraversal<'a> {
         Ok(())
     }
 
+    fn check_multiget_queries(
+        &mut self,
+        op: &'a Positioned<OperationDefinition>,
+    ) -> ServerResult<()> {
+        for item in &op.node.selection_set.node.items {
+            self.traverse_multiget_queries(item)?;
+        }
+        Ok(())
+    }
+
+    fn traverse_multiget_queries(&mut self, item: &'a Positioned<Selection>) -> ServerResult<()> {
+        match &item.node {
+            Selection::Field(f) => {
+                let name = &f.node.name.node;
+                if name.starts_with(MULTI_GET_OBJECTS) {
```

Comment:
The idea was for support in the query limits checker to be fully generic, so that in future, we could add multi gets without making further changes here (so we wouldn't forget to add them). To achieve that we should look for names that start with "multiGet".

Reply:
Will we want to have a limit for all mutli get queries, or we will be doing different limits for different kind of multi get queries?

Reply:
Hmm, good point -- we did move to supporting different limits for different multi-gets, which makes the generic solution more difficult. So the suggestion here changes:

- We should have a version of this check that tests for exactly `multiGetObjects` (i.e. not the prefix).
- We need to think about how we detect a mistake on our part where we expose a multi-get interface and don't appropriately bound it (the answer to this could just be "we need to be vigilant", but I want to make sure there isn't something better we could be doing here).

Longer term (out of scope for this PR) we are going to need to do something similar for pages, where we start supporting different limits for specific paginated endpoints, but that task is slightly easier because we can always fallback on a catch-all limit -- maybe we should do that here too?

Reply:
Oh actually -- I think it's still possible to make this generic, because the only thing that's specific to the kind of multi-get is the upperbound test, which we agreed lower down would be tested for in the resolver for the field, and not in the query limits checker.